cookbooks/bcpc-hadoop/recipes/namenode_master.rb

```diff
@@ -72,12 +72,14 @@
   not_if { node[:bcpc][:hadoop][:mounts].any? { |d| File.exists?("/disk/#{d}/dfs/nn/current/VERSION") } }
 end
 
+zks = node[:bcpc][:hadoop][:zookeeper][:servers].map{|zkh| "#{zkh}:#{node[:bcpc][:hadoop][:zookeeper][:port]}"}.join(",")
+
 bash "format-zk-hdfs-ha" do
   code "yes | hdfs zkfc -formatZK"
   action :run
   user "hdfs"
   notifies :restart, "service[generally run hadoop-hdfs-namenode]", :delayed
-  not_if { zk_formatted? }
+  not_if { znode_exists?("/hadoop-ha/#{node.chef_environment}",zks) }
```

Comment:
This won't work; and this will make it appear as though the namenode is unreachable through an amazingly painful failure process (as I've recently found on a few clusters). So, `zks` is of the form:
````
[{"hostname"=>"cluster-r1n02.example.com", "node_number"=>"9"}, {"hostname"=>"cluster-r1n03.example.com", "node_number"=>"10"}, {"hostname"=>"cluster-r1n06.example.com", "node_number"=>"13"}]
````
However, the function `znode_exists?` expects a data structure of the form `
+zks = node[:bcpc][:hadoop][:zookeeper][:servers].map{|zkh| "#{zkh[:hostname]}:#{node[:bcpc][:hadoop][:zookeeper][:port]}"}.join(",")`